### PR TITLE
Error standardization - Secrets API

### PR DESCRIPTION
**Pull request by yanleiwang**
# Description

## Issue reference

Please reference the issue this PR will close: #8209 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [ ] Code compiles correctly
- [ ] Created/updated tests
- [ ] Unit tests passing
- [ ] End-to-end tests passing
- [ ] Extended the documentation / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_
- [ ] Specification has been updated / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_
- [ ] Provided sample for the feature / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by artursouza** [comment]
Let's merge this one first: https://github.com/dapr/dapr/pull/8192

File: pkg/api/universal/secrets.go [hunk 1]
```diff
@@ -17,9 +17,10 @@ import (
 	"context"
 	"time"
 
+	apierrors "github.com/dapr/dapr/pkg/api/errors"
```
Comment (cicoyle) [inline pkg/api/universal/secrets.go:20]: can you move this down into the section with all the other `"github.com/dapr/dapr...` imports?

File: pkg/api/universal/secrets_test.go [hunk 2]
```diff
@@ -18,14 +18,15 @@ import (
 	"testing"
 	"time"
 
+	"github.com/dapr/dapr/pkg/api/errors"
```
Comment (cicoyle) [inline pkg/api/universal/secrets_test.go:21]: same here

File: pkg/api/errors/secret.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+func (s *SecretStoreError) NotFound() error {
+	msg := fmt.Sprintf("secret store %s not found", s.name)
+	s.skipResourceInfo = true
+	return s.build(
+		errors.NewBuilder(
+			codes.InvalidArgument,
+			http.StatusUnauthorized,
+			msg,
+			"ERR_STORE_NOT_FOUND",
```
Comment (cicoyle) [inline pkg/api/errors/secret.go:59]: This is the legacy tag, we want to keep this the same to not break users
```suggestion
			"ERR_SECRET_STORE_NOT_FOUND",
```

File: tests/integration/suite/daprd/hotreload/selfhosted/secret.go [hunk 4]
```diff
@@ -417,7 +417,7 @@ func (s *secret) readExpectError(t *testing.T, ctx context.Context, compName, ke
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, getURL, nil)
 	require.NoError(t, err)
 	s.doReq(t, req, expCode, fmt.Sprintf(
-		`\{"errorCode":"(ERR_SECRET_STORES_NOT_CONFIGURED|ERR_SECRET_STORE_NOT_FOUND)","message":"(secret store is not configured|failed finding secret store with key %s)"\}`,
+		`\{"errorCode":"(ERR_SECRET_STORES_NOT_CONFIGURED|ERR_STORE_NOT_FOUND)","message":"(secret store is not configured|secret store %s not found)","details":\[\{"@type":"type\.googleapis\.com/google\.rpc\.ErrorInfo","domain":"dapr\.io","metadata":null,"reason":"(DAPR_SECRET_NOT_CONFIGURED|DAPR_SECRET_NOT_FOUND)"\}\]\}`,
```
Comment (cicoyle) [inline tests/integration/suite/daprd/hotreload/selfhosted/secret.go:420]:
```suggestion
		`\{"errorCode":"(ERR_SECRET_STORES_NOT_CONFIGURED| ERR_SECRET_STORE_NOT_FOUND)","message":"(secret store is not configured|secret store %s not found)","details":\[\{"@type":"type\.googleapis\.com/google\.rpc\.ErrorInfo","domain":"dapr\.io","metadata":null,"reason":"(DAPR_SECRET_NOT_CONFIGURED|DAPR_SECRET_NOT_FOUND)"\}\]\}`,
```

File: tests/integration/suite/daprd/secret/grpc/errors.go [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/file"
+	procdaprd "github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/framework/process/secretstore"
+	"github.com/dapr/dapr/tests/integration/framework/process/secretstore/localfile"
+	"github.com/dapr/dapr/tests/integration/framework/socket"
+	"github.com/dapr/dapr/tests/integration/suite"
+	kitErrors "github.com/dapr/kit/errors"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
```
Comment (cicoyle) [inline tests/integration/suite/daprd/secret/grpc/errors.go:38]: for the imports, can you put them into 3 groups:

first section: standard lib imports
empty line
second section: non-standard lib imports & non-dapr imports
empty line
thrid section: dapr imports

**Review by cicoyle — COMMENTED** [review]
Thanks for your contributions @yanleiwang 🎉 this is generally looking good, just a few nits. Super appreciate the integration test coverage 🙏🏻

**Comment by yanleiwang** [comment]
> Thanks for your contributions @yanleiwang 🎉 this is generally looking good, just a few nits. Super appreciate the integration test coverage 🙏🏻

have done the changes suggested. Thank you !

**Pull request closed without merging**